src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration.Tests/MockHttpClientTransport.cs

```diff
@@ -162,6 +144,7 @@ protected override void WriteResponseCore(HttpResponseMessage response)
                 bathItems.Items.Add(KeyValues[itemIndex++]);
             }
             string json = JsonConvert.SerializeObject(bathItems).ToLowerInvariant();
+            json = json.Replace("etag\":{}", "etag\":\"c3c231fd-39a0-4cb6-3237-4614474b92c1\"");
```

Comment:
@jamesnk, we know we can write a custom converter to serialize our ETag type, but maybe there is a better way. Custom  converter seems like a heavy hammer for something that amounts to struct wrapping a string. Is there some attribute or interface to tell JSON.NET to simply call ToString on this type when serializing?

Reply:
I think a a TypeConverter that supports converting to a string will work.

Reply:
@KrzysztofCwalina the plan is still to remove the dependency on JSON.NET, right?
Should we avoid adding more to it? I'm not addressing batches in this PR, but if the replacement I have for now is not enough, I could always do that and competitively eliminate the dependency

Reply:
I think your workaround is ok, if we will remove the dependency relatively soon. But I think it should not linger in the code as it's a bit of a hack :-)

Reply:
lol. it is :( . Once I am done standardizing the API to match the other languages, I will work on this. (Issue #5383 )